### ISSUE TITLE
Unable to reproduce？？？

### ISSUE THREAD

**Issue by haiduo** [timinar/BabyLlama#3]
Hello, I follow the configuration of your paper and used GPT-2 (705M, eval_loss: 3.75) and LLaMA (360M, eval_loss: 3.746) to distill babyLLamA (58M), but I get eval_loss: 27.985. The gap is so big. What is the reason? 

In addition, I trained babyLLamA (58M) from scratch alone and got eval_loss: 4.023. The following is my eval loss process, for the above four models (GPT-2(705M, eval_loss:3.75), LLaMA(360M, eval_loss:3.746), babyLLamA(58M), distill-babyLLamA(58M)):
![image](https://github.com/timinar/BabyLlama/assets/35062202/9a52a36b-60af-4d11-8655-2c1785ba1f59)
![image](https://github.com/timinar/BabyLlama/assets/35062202/60311708-5a65-4ee5-804d-14ffe72b82e9)
![image](https://github.com/timinar/BabyLlama/assets/35062202/db9fda5a-f0bc-45db-8ce2-224fa6faa15f)
![image](https://github.com/timinar/BabyLlama/assets/35062202/b7e409cd-2737-4443-ab86-a5f188960c77)

Others results:
![image](https://github.com/timinar/BabyLlama/assets/35062202/10ec5b0e-67af-4423-9054-e62223fa96f9)

**Comment by timinar** [comment]
Hello! This is expected. In the distillation pertaining, we use the loss that combines the usual cross-entropy with a KL divergence between the student's and teachers' probability distributions $L = \alpha L_{CE} + (1-\alpha) L_{KL}$. This second term is term is pretty large, so the resulting loss is completely off compared to the cross-entropy one. Yet, the model trains. Ideally, we should have computed and logged the cross-entropy on eval. Alternatively, one can load the trained model and eval it on the same evaluation dataset.

**Comment by haiduo** [comment]
> ello! This is expected. In the distillation pertaining, we use the loss that combines the usual cross-entropy with a KL divergence between the student's and teachers' probability distributions L=αLCE+(1−α)LKL. This second term is term is pretty large, so the resulting loss is completely off compared to the cross-entropy one. Yet, the model trains. Ideally, we should have computed and logged the cross-entropy on eval. Alternatively, one can load the trained model and eval it on the same evaluation dataset.

Okay, thank you for your reply. I will try their actual test level on the evaluation dataset later.

**Comment by shawnricecake** [comment]
Hi,

How can I get the babyllama dataset?

Is there any code for evaluation on other datasets as reported in the paper?

Thanks

**Comment by JLTastet** [comment]
Sorry for the late reply.

The BabyLM dataset can be downloaded from the webpage of the challenge: https://babylm.github.io/

We haven’t evaluated the model on other datasets, but since we rely on the HuggingFace Transformers library, I would expect it to be widely-compatible with eval frameworks.

**Comment by shawnricecake** [comment]
> Sorry for the late reply.
> 
> The BabyLM dataset can be downloaded from the webpage of the challenge: https://babylm.github.io/
> 
> We haven’t evaluated the model on other datasets, but since we rely on the HuggingFace Transformers library, I would expect it to be widely-compatible with eval frameworks.

Thanks for your reply! That truly helps!

**Comment by shawnricecake** [comment]
@haiduo Hi, do you mind share that how you get the eval loss? thanks!
